xds/internal/server/listener_wrapper.go

```diff
@@ -414,19 +414,33 @@ type ldsWatcher struct {
 	name   string
 }
 
-func (lw *ldsWatcher) OnUpdate(update *xdsresource.ListenerResourceData, onDone xdsresource.OnDoneFunc) {
+func (lw *ldsWatcher) OnResourceChanged(update *xdsresource.ResourceDataOrError, onDone xdsresource.OnDoneFunc) {
 	defer onDone()
 	if lw.parent.closed.HasFired() {
-		lw.logger.Warningf("Resource %q received update: %#v after listener was closed", lw.name, update)
+		if update.Err != nil {
+			lw.logger.Warningf("Resource %q received err: %#v after listener was closed", lw.name, update.Err)
+		} else {
+			lw.logger.Warningf("Resource %q received update: %#v after listener was closed", lw.name, update)
+		}
 		return
 	}
 	if lw.logger.V(2) {
-		lw.logger.Infof("LDS watch for resource %q received update: %#v", lw.name, update.Resource)
+		if update.Err != nil {
+			lw.logger.Infof("LDS watch for resource %q received error: %#v", lw.name, update.Err)
+		} else {
+			u := update.Data.(*xdsresource.ListenerResourceData)
+			lw.logger.Infof("LDS watch for resource %q received update: %#v", lw.name, u.Resource)
+		}
+	}
```

Comment:
Same comment as in the other file. Please add logs along with the code that handles the error or the update instead of having a separate block for logging.